frontend/src/metabase/common/hooks/use-setting/use-setting.ts

```diff
@@ -1,9 +1,23 @@
-import { useSelector } from "metabase/lib/redux";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import type { UpdateUserSettingOptions } from "metabase/redux/settings";
+import { updateUserSetting } from "metabase/redux/settings";
 import { getSetting } from "metabase/selectors/settings";
-import type { Settings } from "metabase-types/api";
+import type { Settings, UserSettings } from "metabase-types/api";
 
 export const useSetting = <SettingName extends keyof Settings>(
   settingName: SettingName,
 ) => {
   return useSelector(state => getSetting(state, settingName));
 };
+
+export const useUserSetting = <T extends keyof UserSettings>(
```

Comment:
I really think that this hook needs up update the store upon completion. A quick screencap to show what I mean: 
![chrome_74FQjBtvP5](https://github.com/metabase/metabase/assets/1328979/8c28e3ce-7aad-4195-97df-442b6fd5d28a)